src/main/java/com/microsoft/azure/functions/worker/broker/FactoryJavaMethodExecutor.java

```diff
@@ -0,0 +1,23 @@
+package com.microsoft.azure.functions.worker.broker;
+
+import com.microsoft.azure.functions.rpc.messages.BindingInfo;
+import com.microsoft.azure.functions.worker.WorkerLogManager;
+import com.microsoft.azure.functions.worker.description.FunctionMethodDescriptor;
+import com.microsoft.azure.functions.worker.reflect.ClassLoaderProvider;
+import org.apache.commons.lang3.SystemUtils;
+
+import java.net.MalformedURLException;
+import java.util.Map;
+
+public class FactoryJavaMethodExecutor {
+    public JavaMethodExecutor getJavaMethodExecutor(FunctionMethodDescriptor descriptor, Map<String, BindingInfo> bindings, ClassLoaderProvider classLoaderProvider)
+            throws MalformedURLException, ClassNotFoundException, NoSuchMethodException {
+        if(SystemUtils.IS_JAVA_1_8) {

```

Comment:
Not needed to use SystemUtils here. You can replace it with a try catch (catching Throwable) using

boolean isJava8 = true;
try {
  Runtime.Version.feature()
  isJava8 = false;
}
catch(Throwable t) {
}

Reply:
I believe Runtime.Version is introduced in java 9, we are still code compatible with java 8. https://stackoverflow.com/questions/2591083/getting-java-version-at-runtime

Reply:
The reason I used SystemUtils.IS_JAVA_1_8, it is more cleaner to hide the functionality, as it is calling at the end java.specification.version

Reply:
I mentioned this here in case you would want to get rid of the use of SystemUtils. Note this suggestion will work on both versions (it does require Java 9+ to do the compile). Maybe an simple opportunity target if 2 branches are going to be maintained.